h-include.js

```diff
@@ -41,35 +41,64 @@ var hinclude;
   hinclude = {
     classprefix: "include_",
 
-    show_content: function (element, req) {
-      var i, include, message, fragment = element.getAttribute('fragment') || 'body';
-      if (req.status === 200 || req.status === 304) {
-        var container = document.implementation.createHTMLDocument().documentElement;
-        container.innerHTML = req.responseText;
+    create_container: function (req) {
+      var container = document.implementation.createHTMLDocument().documentElement;
+      container.innerHTML = req.responseText;
+
+      return container;
+    },
+    check_recursion: function (container, element) {
```

Comment:
Yes, it can be optimised. However, in that case we would need create another DOM container to find the h-include elements and I'd like to avoid that.

From my perspective, this check is mostly to prevent something very bad and I'd rather optimise for small code size and maybe let a few ajax requests "slip through". How does that sound?